interfaces/apparmor/backend.go

```diff
@@ -109,6 +109,18 @@ func (b *Backend) Initialize() error {
 		logger.Noticef("snapd enabled NFS support, additional implicit network permissions granted")
 	}
 
+	// Check if '/' is on overlayfs. If so, add the necessary rules for
+	// upperdir and allow snap-confine to work.
+	if overlayRoot, err := osutil.IsRootOverlay(); err != nil {
+		logger.Noticef("cannot determine if / on overlay: %v", err)
+	} else if overlayRoot {
+		policy["overlay-root"] = &osutil.FileState{
+			Content: []byte(overlayRootSnippet),
+			Mode:    0644,
+		}
+		logger.Noticef("snapd enabled / on overlay support, additional upperdir permissions granted")
```

Comment:
`/` may be misread a "or", how about `root filesystem on overlayfs, additional upperdir permissions granted`?

Reply:
Done